cmake/kokkos_install.cmake

```diff
@@ -0,0 +1,67 @@
+
+SET(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for libraries")
+SET(INSTALL_BIN_DIR bin CACHE PATH "Installation directory for executables")
+SET(INSTALL_INCLUDE_DIR ${KOKKOS_HEADER_DIR} CACHE PATH
+  "Installation directory for header files")
+
+#Set all the variables needed for kokkosConfig.cmake
+GET_PROPERTY(KOKKOS_PROP_LIBS GLOBAL PROPERTY KOKKOS_LIBRARIES_NAMES)
+SET(KOKKOS_LIBRARIES ${KOKKOS_PROP_LIBS})
+
+SET(DEF_INSTALL_CMAKE_DIR)
+IF(WIN32 AND NOT CYGWIN)
+  LIST(APPEND DEF_INSTALL_CMAKE_DIR CMake)
+ELSE()
+  LIST(APPEND DEF_INSTALL_CMAKE_DIR lib/CMake/Kokkos)
```

Comment:
Why do we bother with `<prefix>/lib/CMake`?

Reply:
This is where the config was getting installed in the previous CMake. For the time being, I decide to still install the config file to the old location to limit the chaos.  This can be removed? But I don't know how many users in the wild will be frustrated by KokkosConfig.cmake having moved to a new location.

Reply:
I am also installing to the lowercase `/lib/cmake` as well as the preferred location.

Reply:
Oh I see.  You are still exporting that `kokkos` target w/o the namespace.

Reply:
And now we are also installing it (the old `kokkos` target w/o namespace) at the right place...

Reply:
I think this is a bad idea.
I would only install the new `Kokkos::kokkos` target at the right place (`<prefix>/lib/cmake/Kokkos`).
And (if we want to provide it at all) the old `kokkos` one where it use to be (`<prefix>/lib/Kokkos/CMake`).

Reply:
Sure, that might make the most sense. We can install the namespaced targets to the "correct" location and the deprecated targets to the "wrong" location.

Reply:
You take care of it or me?

Reply:
Maybe easiest if you just make the change when you update the `KOKKOS_ASSERT` stuff?

Reply:
This is somewhat broken on Mac now. Darwin is case-insensitive so both sets of files get installed in the "same" place.

Reply:
Actually a lot broken. The deprecated (non-namespaced) names overwrite the new Kokkos:: names. This is what I'm going to do (for now) since I have to repair this so I can keep working on Spack. If Darwin, I will NOT install the deprecated targets. If we decide later that we want the deprecated targets for Mac, a more significant refactor will be needed.